### PR TITLE
feat: publish metrics to sonar run for ux-test-platform

### PR DESCRIPTION
**Pull request by umahaea**
This will technically be not merged into "rebrand" branch. We will maintain "sonar-full-coverage" branch for the sonar activity and continuously rebase it with "rebrand".

Created a PR for feedback and review comments only.

### CODE REVIEW

File: src/main/shell_scripts/components.sh [hunk 1]
```diff
@@ -186,11 +200,27 @@ npm install &>/dev/null
...
+instrument_file(){
+sed -i -e 's/\/\/# sourceMappingURL/sourceMappingURL/g' ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/$1/$2.js
+mv ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/$1/$2.js ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/$1/$2.test.js
+js-beautify ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/$1/$2.test.js >> ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/$1/$2.js
+nyc instrument ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/$1/$2.js >> ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/$1/$2-instrumented.js
+}
+
+instrument_file_elements-sdk(){
+sed -i -e 's/\/\/# sourceMappingURL/sourceMappingURL/g' ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/elementsSDK/functional/jsfiles/$1.js
+mv ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/elementsSDK/functional/jsfiles/$1.js ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/elementsSDK/functional/jsfiles/$1.test.js
```
Comment (udhadpa) [inline src/main/shell_scripts/components.sh]: liked this. Not touching the original file

File: src/main/shell_scripts/components.sh [hunk 2]
```diff
@@ -186,11 +200,27 @@ npm install &>/dev/null
...
 cp ~/build/Pearson-Higher-Ed/ux-test-platform/coach-mark/build/dev.coach-mark.js ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/coachMark/
+instrument_file coachMark dist.coach-mark
+instrument_file coachMark dev.coach-mark
 cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/coach-mark/node_modules/pearson-elements/dist/fonts ~/build/Pearson-Higher-Ed/ux-test-platform/
 cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/coach-mark/node_modules/pearson-elements/dist/icons ~/build/Pearson-Higher-Ed/ux-test-platform/
 cp ~/build/Pearson-Higher-Ed/ux-test-platform/coach-mark/node_modules/pearson-elements/dist/css/elements.css ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/css/coachMark/
 }
 
+instrument_file(){
+sed -i -e 's/\/\/# sourceMappingURL/sourceMappingURL/g' ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/standAlone/jsfiles/$1/$2.js
```
Comment (udhadpa) [inline src/main/shell_scripts/components.sh]: nice one, after you explained :) 💯

**Comment by udhadpa** [comment]
Superb work Eajaz. This looks awesome.. 
Highly impressive, more power to you 🥇 💯 👍

**Comment by umahaea** [comment]
Thanks Payal.. Nice comments  :)

**Pull request closed without merging**